src/neuralnet/researcher.cpp

```diff
@@ -180,6 +180,61 @@ const Project* ResolveWhitelistProject(
     return nullptr;
 }
 
+//!
+//! \brief Represents a Gridcoin pool that stakes on behalf of its users.
+//!
+//! The wallet uses these entries to detect when BOINC is attached to a pool
+//! account so that it can provide more useful information in the UI.
+//!
+class MiningPool
+{
+public:
+    MiningPool(const Cpid cpid, std::string m_name, std::string m_url)
+        : m_cpid(cpid), m_name(std::move(m_name)), m_url(std::move(m_url))
+    {
+    }
+
+    MiningPool(const std::string& cpid, std::string m_name, std::string m_url)
+        : MiningPool(Cpid::Parse(cpid), std::move(m_name), std::move(m_url))
+    {
+    }
+
+    Cpid m_cpid;        //!< The pool's external CPID.
+    std::string m_name; //!< The name of the pool.
+    std::string m_url;  //!< The pool's website URL.
+};
+
+//!
+//! \brief The set of known Gridcoin pools.
+//!
+//! TODO: In the future, we may add a contract type that allows pool operators
+//! to register a pool via the blockchain. The static list gets us by for now.
+//!
+const MiningPool g_pools[] = {
+    { "7d0d73fe026d66fd4ab8d5d8da32a611", "grcpool.com", "https://grcpool.com/" },
+    { "a914eba952be5dfcf73d926b508fd5fa", "grcpool.com-2", "https://grcpool.com/" },
+    { "163f049997e8a2dee054d69a7720bf05", "grcpool.com-3", "https://grcpool.com/" },
+    { "326bb50c0dd0ba9d46e15fae3484af35", "Arikado", "https://gridcoinpool.ru/" },
```

Comment:
Rather than hard-coding, lets make a protocol entry for this, which we can then replace with a self-service contract in the future...

Reply:
I thought about that. In light of the pending release, I wanted to spend as little time on this change as possible, so I went for the static list. Using the protocol entries from the `AppCache` requires thread synchronization and hooks for state changes, and it's slower to fetch the data and parse out the CPIDs from the strings. We're doing that for the team whitelist, so maybe it's easy enough. I'd rather work on this after everything else is finished--we can switch to using protocol entries in a leisure release.

Reply:
Ok.